lib/core/base/audit.js

```diff
@@ -327,58 +436,79 @@ Audit.prototype.addCheck = function(spec) {
  */
 Audit.prototype.run = function(context, options, resolve, reject) {
 	'use strict';
+
 	this.normalizeOptions(options);
 
 	axe._selectCache = [];
-	var q = axe.utils.queue();
-	this.rules.forEach(function(rule) {
-		if (axe.utils.ruleShouldRun(rule, context, options)) {
-			if (options.performanceTimer) {
-				var markEnd = 'mark_rule_end_' + rule.id;
-				var markStart = 'mark_rule_start_' + rule.id;
-				axe.utils.performanceTimer.mark(markStart);
+
+	// get a list of rules to run NOW vs. LATER (later are preload assets dependent rules)
+	const allRulesToRun = getRulesToRun(this.rules, context, options); //TODO:JEY - test
+	const runNowRules = allRulesToRun.now;
+	const runLaterRules = allRulesToRun.later;
+
+	// init a NOW queue for rules to run immediately
+	const nowQueue = axe.utils.queue();
+
+	// construct can run NOW rules into NOW queue
+	runNowRules.forEach(rule => {
+		const deferredRule = getDefferedRule(rule, context, options);
+		nowQueue.defer(deferredRule);
+	});
+
+	// defer preload into the NOW queue
+	// only if there are rules that need assets
+	if (runLaterRules.length) {
+		const preloader = axe.utils.preload(options);
```

Comment:
You're not handling preloader timeout / throwing. Preload failing should not cause the entire run to fail. If preloader times out, we need the rules that require preload to resolve to turn out incomplete results. Please make sure there are tests for this, and lets look at if we need one incomplete result for a failed preload, or if we need to run the selector anyway, just in case.